ServerCore/PuzzleStateHelper.cs

```diff
@@ -34,60 +35,29 @@ public static IQueryable<PuzzleStatePerTeam> GetFullReadOnlyQuery(PuzzleServerCo
             }
 
 #pragma warning disable IDE0031 // despite the compiler message, "teamstate?.UnlockedTime", etc does not compile here
 
             if (puzzle != null)
             {
-                IQueryable<Team> teams = context.Teams.Where(t => t.Event == eventObj);
-                IQueryable<PuzzleStatePerTeam> states = context.PuzzleStatePerTeam.Where(state => state.Puzzle == puzzle);
-
-                return from t in teams
-                       join state in states on t.ID equals state.TeamID into tmp
-                       from teamstate in tmp.DefaultIfEmpty()
-                       select new PuzzleStatePerTeam
-                       {
-                           Puzzle = puzzle,
-                           PuzzleID = puzzle.ID,
-                           Team = t,
-                           TeamID = t.ID,
-                           UnlockedTime = teamstate == null ? null : teamstate.UnlockedTime,
-                           SolvedTime = teamstate == null ? null : teamstate.SolvedTime,
-                           Printed = teamstate == null ? false : teamstate.Printed,
-                           Notes = teamstate == null ? null : teamstate.Notes,
-                           IsEmailOnlyMode = teamstate == null ? false : teamstate.IsEmailOnlyMode
-                       };
+                return context.PuzzleStatePerTeam.Where(state => state.Puzzle == puzzle);
             }
 
             if (team != null)
             {
-                IQueryable<Puzzle> puzzles = author == null ? context.Puzzles.Where(p => p.Event == eventObj) : UserEventHelper.GetPuzzlesForAuthorAndEvent(context, eventObj, author);
-                IQueryable<PuzzleStatePerTeam> states = context.PuzzleStatePerTeam.Where(state => state.Team == team);
-
-                return from p in puzzles
-                       join state in states on p.ID equals state.PuzzleID into tmp
-                       from teamstate in tmp.DefaultIfEmpty()
-                       select new PuzzleStatePerTeam
-                       {
-                           Puzzle = p,
-                           PuzzleID = p.ID,
-                           Team = team,
-                           TeamID = team.ID,
-                           UnlockedTime = teamstate == null ? null : teamstate.UnlockedTime,
-                           SolvedTime = teamstate == null ? null : teamstate.SolvedTime,
-                           Printed = teamstate == null ? false : teamstate.Printed,
-                           Notes = teamstate == null ? null : teamstate.Notes,
-                           IsEmailOnlyMode = teamstate == null ? false : teamstate.IsEmailOnlyMode
-                       };
+                if (author != null)
+                {
+                    return from state in context.PuzzleStatePerTeam
+                           join auth in context.PuzzleAuthors on state.PuzzleID equals auth.PuzzleID
+                           where state.Team == team
```

Comment:
where does this filter to the author? Since "author" is not present in the query I suspect it doesn't.

Reply:
Good catch, fixed